src/Voron/Data/Lookups/ILookupKey.cs

```diff
@@ -25,7 +25,5 @@ public interface ILookupKey
 
     void OnKeyRemoval<T>(Lookup<T> parent) where T : struct, ILookupKey;
     
-    void IncreaseReferenceCount<T>(Lookup<T> parent) where T : struct, ILookupKey;
```

Comment:
This effectively reverts https://github.com/ravendb/ravendb/pull/17506. We have decided to approach it differently,

Instead of incrementing the term reference count before doing internal operations (like moving all entries from one leaf to another and then freeing / removing old page), we'll avoid decrements when doing internal tree adjustments (during splits or rebalancing).